### PR TITLE
Move all tests from travis to drone

### PR DESCRIPTION
**Pull request by paurakhsharma**
# Description
Move all tests from `travis.yml` to `drone.yml`

Related issue: #26

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -208,7 +205,40 @@ matrix:
...
 
+    - PHP_VERSION: 5.6
+      OC_VERSION: daily-stable10-qa
+      TEST_SUITE: phpunit
+      DB_TYPE: sqlite
+      DB_NAME: owncloud
+      NEED_SERVER: true
+      NEED_INSTALL_APP: true
+
+    - PHP_VERSION: 5.6
```
Comment (phil-davis) [inline .drone.yml]: This entry looks the same as the entry above.

File: .drone.yml [hunk 2]
```diff
@@ -68,6 +68,7 @@ pipeline:
     pull: true
     environment:
       - PHP_VERSION=${PHP_VERSION}
+      - COVERAGE=${COVERAGE}
     commands:
       - cd /var/www/owncloud/apps/brute_force_protection/tests
       - phpdbg -d memory_limit=4096M -rr ../../../lib/composer/phpunit/phpunit/phpunit --configuration phpunit.xml
```
Comment (phil-davis) [inline .drone.yml]: This can be adjusted to to the new ``test-php-unit`` etc in the make file.

**Comment by paurakhsharma** [comment]
@phil-davis, codecov is decreased by 70%, is this an issue or is it expected?

**Comment by phil-davis** [comment]
@DeepDiver1975 please review. If you think it is OK, the force merge.

**Comment by DeepDiver1975** [comment]
There is something wrong with coderage upload .... https://codecov.io/gh/owncloud/brute_force_protection/pull/46/tree

**Comment by DeepDiver1975** [comment]
```
phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "./vendor/bin/phpunit" --configuration ./tests/phpunit.xml --testsuite unit
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

Runtime:       PHPDBG 7.1.20-1+ubuntu16.04.1+deb.sury.org+1
Configuration: /var/www/owncloud/apps/brute_force_protection/tests/phpunit.xml



Time: 168 ms, Memory: 14.00MB

No tests executed!

Generating code coverage report in Clover XML format ... done

**Comment by paurakhsharma** [comment]
@DeepDiver1975,  I have made some changes, unit tests are being executed now. Please review this again.

File: .drone.yml [hunk 3]
```diff
@@ -95,21 +96,17 @@ pipeline:
...
-#        DB_TYPE: mysql
-#        OC_VERSION: daily-master-qa
+  codecov:
+    image: plugins/codecov:2
+    secrets: [codecov_token]
+    pull: true
+    files:
+    - '*.xml'
+    paths:
+    - tests/clover.xml
```
Comment (DeepDiver1975) [inline .drone.yml]: tests/output/closer.xml

File: Makefile [hunk 4]
```diff
@@ -106,30 +126,20 @@ else
...
 	tar -czf $(appstore_package_name).tar.gz -C $(appstore_package_name)/../ $(app_name)
 
-.PHONY: test-php-style
-test-php-style:
-	composer install
-	vendor/bin/php-cs-fixer fix -v --diff --diff-format udiff --dry-run --allow-risky yes
+.PHONY: test-php-unit
+test-php-unit:             ## Run php unit tests
+test-php-unit: vendor/bin/phpunit
+	$(PHPUNIT) --configuration ./tests/phpunit.xml --testsuite unit
```
Comment (DeepDiver1975) [inline Makefile]: move phpunit.xml into repo root

File: tests/phpunit.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
 	 strict="true"
```
Comment (DeepDiver1975) [inline tests/phpunit.xml]: boostrap file above might be dropped and core boostrap.php can be used

**Comment by karakayasemi** [comment]
please also update badge in the readme. Thanks

File: .drone.yml (hunk 3 above)
Comment (phil-davis) [reply .drone.yml]:
```
tests/output/clover.xml
```
but I am sure you know that @paurakhsharma

**Comment by paurakhsharma** [comment]
@phil-davis @DeepDiver1975 @karakayasemi, I have made the necessary changes. Please review it again.

File: .drone.yml [hunk 6]
```diff
@@ -95,21 +96,17 @@ pipeline:
...
-#        TEST_SUITE: phpunit
-#        PHP_VERSION: 7.2
-#        DB_TYPE: mysql
-#        OC_VERSION: daily-master-qa
+  codecov:
+    image: plugins/codecov:2
+    secrets: [codecov_token]
+    pull: true
+    files:
+    - '*.xml'
```
Comment (DeepDiver1975) [inline .drone.yml]: we need files and paths? looks wrong ...

File: README.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
 # Brute Force Protection App
-[![Build Status](https://travis-ci.org/owncloud/brute_force_protection.svg?branch=master)](https://travis-ci.org/owncloud/brute_force_protection)
+[![Build Status](https://drone.owncloud.com/api/badges/owncloud/brute_force_protection/status.svg)](https://drone.owncloud.com/owncloud/brute_force_protection)
```
Comment (DeepDiver1975) [inline README.md]: Please add codecov badge as well - THX a lot!

File: .drone.yml (hunk 6 above)
Comment (paurakhsharma) [reply .drone.yml]: Oh, I copied it from `password_policy`, I will only use files then.

File: README.md (hunk 7 above)
Comment (paurakhsharma) [reply README.md]: @DeepDiver1975, I have made the changes, please review it again.

**Comment by patrickjahns** [comment]
@DeepDiver1975 @phil-davis @paurakhsharma 
Please keep an eye out when using `composer-bin` to not commit the `composer.lock` files
see https://github.com/owncloud/brute_force_protection/pull/46/files#diff-b5b96552c279a6308c6e64a8718d2126

Reason being - that we do want to use the latest tooling to be available automatically and not pin the tools to a version (yet). Only if/when we encounter troubles that a tool would break, we should pin it to a different version

**Pull request merged**